### DISCUSSION TITLE
talosctl recover after VMs restart

### DISCUSSION THREAD

**Discussion by skillpoint-dev** [siderolabs/talos#3982] [Q&A]
Here is what I have written down for what to do when I restart the VMs where I have Talos running(I got this info in the Talos Slack channel)...

> As of "Talos OS 0.8.0-alpha3(k8s 1.20.0)" there is a "Node Affinity"-Bug in k8s for single control plane node clusters.
> After restart the pods wont start correctly.
>
> Remedy
>
> `talosctl recover --nodes [TALOS-CONTROL-PLANE-IP]`
>
> or
>
> `talosctl recover --nodes [TALOS-CONTROL-PLANE-IP] --source etcd`

Unfortunately this doesn't work anymore but I still face the problem that most of the applications don't recover from a VM restart.

Is there anything that is now in place instead of "talosctl recover" - I only found the disaster recovery page and this seems the wrong place to look for that.

**Comment by andrewrynhard** [comment]
Which version of Talos? The `recover` command worked in the self-hosted Kubernetes format. That currently isn't the way Talos deploys things so that command is a noop on clusters that aren't self-hosted.

**Reply by skillpoint-dev** [reply]
Last try of a recovery after a VM restart was with 0.11.2 but I don't remember at which version the suggested remedy of the problem above was not working anymore.

**Reply by smira** [reply]
Was the cluster bootstrapped on Talos <= 0.8. If not, it's not using self-hosted control plane, so `talosctl recover` won't work. 

If the cluster was bootstrapped on Talos <= 0.8, it should be converted to the new control plane format: https://www.talos.dev/docs/v0.11/guides/converting-control-plane/
